### PR TITLE
Update htmlnano config

### LINKED ISSUE #6594: htmlnano configuration files
More a note than anything, but there's a commit on htmlnano adding native configuration files (https://github.com/posthtml/htmlnano/commit/2c03b10d2dfd606c828f66a07ccfb29871afae1c). This hasn't yet been released.

They're slightly different to Parcel's config files in that `preset` key has a specific meaning.

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

Change htmlnano config filepaths and key in package.json to match the keys/filenames described in the readme.

See https://github.com/posthtml/htmlnano#config

Closes #6594 

Related PR to disable config loading in htmlnano: https://github.com/posthtml/htmlnano/pull/146

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 8.35s | +15.00ms |
| Cached | 431.00ms | -10.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.07s | -16.00ms |
| Cached | 371.00ms | -4.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/304717731333145152)**

File: packages/optimizers/htmlnano/src/HTMLNanoOptimizer.js [hunk 1]
```diff
@@ -13,7 +13,10 @@ export default (new Optimizer({
   async loadConfig({config, options}) {
     let userConfig = await config.getConfigFrom(
       path.join(options.entryRoot, 'index.html'),
-      ['.htmlnanorc', '.htmlnanorc.js'],
+      ['htmlanorc.json', 'htmlnanorc.js'],
```
Comment (devongovett) [inline packages/optimizers/htmlnano/src/HTMLNanoOptimizer.js]: Hmm we've supported .htmlnanorc for a long time and a bunch of projects might use it. Should we still support it in addition to the new ones? Should we deprecate it and have a warning?

Comment (DeMoorJasper) [reply packages/optimizers/htmlnano/src/HTMLNanoOptimizer.js]: We can wait until parcel 3 to remove it if you prefer that, not sure

Comment (devongovett) [reply packages/optimizers/htmlnano/src/HTMLNanoOptimizer.js]: Not sure. Let's discuss with the team. Another option is to detect the file but make it an error so people know to rename it when upgrading.

Comment (thewilkybarkid) [reply packages/optimizers/htmlnano/src/HTMLNanoOptimizer.js]: The README mentions these, but I don't think it's an exhaustive list. (I believe it's all the names that [cosmicconfig](https://github.com/davidtheclark/cosmiconfig) supports.)

Comment (devongovett) [reply packages/optimizers/htmlnano/src/HTMLNanoOptimizer.js]: Ah you're right. According to [this](https://github.com/davidtheclark/cosmiconfig/blob/main/README.md#cosmiconfig) it should be:

* package.json#htmlnano
* `.htmlnanorc`
* `.htmlnanorc.json`
* `.htmlnanorc.yml`
* `.htmlnanorc.yaml`
* `.htmlnanorc.js`
* `.htmlnanorc.cjs`
* `htmlnano.config.js`
* `htmlnano.config.cjs`

File: packages/core/integration-tests/test/cache.js [hunk 2]
```diff
@@ -5343,7 +5343,7 @@ describe('cache', function() {
         entries: ['src/index.html'],
         async setup() {
           await overlayFS.writeFile(
-            path.join(inputDir, '.htmlnanorc'),
+            path.join(inputDir, '.htmlanorc.json'),
```
Comment (thewilkybarkid) [inline packages/core/integration-tests/test/cache.js]: Looks like there’s a few typos.

Comment (DeMoorJasper) [reply packages/core/integration-tests/test/cache.js]: Oh wow that explains the tests failing, thanks for this, totally missed that

Comment (thewilkybarkid) [reply packages/core/integration-tests/test/cache.js]: Easily done. 😅

File: packages/optimizers/htmlnano/src/HTMLNanoOptimizer.js [hunk 3]
```diff
@@ -36,6 +44,10 @@ export default (new Optimizer({
       );
     }
 
+    // $FlowFixMe
+    const presets = htmlnano.presets;
+    const preset =
+      typeof config?.preset === 'string' ? presets[config.preset] : {};
```
Comment (thewilkybarkid) [inline packages/optimizers/htmlnano/src/HTMLNanoOptimizer.js]: There should be a test case for using the `preset` config option, given it has special behaviour.

**Review by devongovett — APPROVED** [review]

**Pull request merged**